Responsive

alfonsogarciacaro opened the pull request:
![nacara](https://user-images.githubusercontent.com/8275461/58216022-dfcf0500-7cfc-11e9-9904-6a0219d23c47.gif)

MangelMaxime commented on scss/nacara.scss:
```diff
@@ -63,3 +63,18 @@ body {
 }
+
+// TODO: Mobile width, is there a variable for this?
+@media (max-width: 768px) {
```
We can avoid this custom CSS by using `is-hidden-mobile` [doc](https://bulma.io/documentation/modifiers/responsive-helpers/#hide) directly on the TOC element

MangelMaxime commented on src/Render.DocPage.fs:
```diff
@@ -9,17 +9,31 @@ open Fable.Import
+          Column.column [ Column.Width (Screen.Desktop, Column.Is2)
+                          Column.Width (Screen.Tablet, Column.Is3)
+                          Column.Width (Screen.Mobile, Column.IsFull)
                           Column.CustomClass "full-height-scrollable-content"
```
Here is the place where we can add `is-hidden-mobile` via `Modifier.IsHidden (Screen.Mobile, true)`

alfonsogarciacaro commented:
Thanks! I've implemented the changes. I also added some fade-in animation for the TOC (fade-out is trickier but we could do it too).

![nacara](https://user-images.githubusercontent.com/8275461/58244778-a2499680-7d53-11e9-94ac-eb45476fef47.gif)

MangelMaxime commented on scss/nacara.scss:
```diff
@@ -63,3 +63,26 @@ body {
+
+#toc-column {
+    // TODO: I'd like to reduce the top margin in mobile
+    // But the jump when changing to content is noticeable
```
I didn't try the code.

But if you want it to be less abrupt you can animate the property.

alfonsogarciacaro commented:
Ok, I used the animation to reduce the top margin for the TOC in mobile mode, though I'm not sure if it looks better :) If the PR is good for you (feel free to make any changes), could you please merge so we can start using this with the Fable docs? Thanks!

alfonsogarciacaro commented:
While testing on mobile I noticed the meta viewport tag wasn't set. I've fixed that as well :)

<img width="455" alt="Screen Shot 2019-05-23 at 10 06 40 PM" src="https://user-images.githubusercontent.com/8275461/58283602-f7fc5e00-7da8-11e9-9d83-7791d24f68d5.png">

whitetigle commented:
Yes it seems sooooooooooooo great! 😊 We want this for fabledoc! we want this for Fabledoc! 😊 😊 😊 😊 😊 😊 😊 😊 😊 😊 😊 😊 😊 😊 😊 😊  (now I'm sure you're convinced to merge this awesome PR!!!)

MangelMaxime commented:
I will take time to review this PR and also change the "toggle button" because it doesn't really look nice ^^

alfonsogarciacaro commented:
@MangelMaxime The PR is growing quite big... that's what happens when you don't merge my PRs quickly :wink:

I needed the responsive change for the Fable docs, and I've added some minor features, like making the watch mode and live server optionals, for better integration with the Fable site build pipeline. I also updated the Fable dependencies and replaced the npm "string" package because it was giving security warnings.

In order to test the tool, I've published the fork with the `@fable-compiler` scope. When you're back it'd be great if you could merge and republish so we can just work with a single version. I hope you're enjoying the holidays btw! :beach:

MangelMaxime commented:
> The PR is growing quite big... that's what happens when you don't merge my PRs quickly 😉

That's why you can create several PR refering each others ;)

Before my holidays, I started working on the next version of Nacara which introduce the layout and plugin system. 

The goal is to make all the rendering process independent of Nacara core to avoid modifying the code of Nacara for adding new user features.

I will make sure to forward the changes from this PR to the next version of Nacara. 

I will also make the appropriate changes to fable-doc repo when published.

alfonsogarciacaro commented:
Chaining PRs is a bit pointless if they're building on each other, because the last PR is going to contain all the commits anyways. Besides, if PR A is not merged, there's no guarantee that A+B+C are going to be merged ;)

Please also note that fable-doc repo is archived now, documentation has already been integrated on the fable-compiler.github.io repository.

MangelMaxime commented:
Chaining PRs are easier to review because you see "one change" at a time :) but I get your point :)

> Please also note that fable-doc repo is archived now, documentation has already been integrated on the fable-compiler.github.io repository.

Ok I take note :)

MangelMaxime commented:
Version `0.2.0` of Nacara will be publish today and it's already support the feature included in this PR.

Thanks for the contribution and sorry for the delay

whitetigle commented:
Thanks 😃

MangelMaxime commented:
@whitetigle @alfonsogarciacaro For info, you can find the list of changes here: https://mangelmaxime.github.io/Nacara/changelog.html
